src/main/java/chess/web/RoomApiController.java

```diff
@@ -0,0 +1,58 @@
+package chess.web;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import chess.service.GameService;
+import chess.service.RoomService;
+import chess.web.dto.BoardDto;
+import chess.web.dto.RoomResponseDto;
+
+@RestController
+@RequestMapping("/api/rooms")
+public class RoomApiController {
+
+	private final RoomService roomService;
+	private final GameService gameService;
+
+	public RoomApiController(RoomService roomService, GameService gameService) {
+		this.roomService = roomService;
+		this.gameService = gameService;
+	}
+
+	@GetMapping
+	public ResponseEntity<List<RoomResponseDto>> loadRooms() {
+		List<RoomResponseDto> rooms = roomService.findAll().stream()
+			.map(room -> new RoomResponseDto(
+				room.getId(),
+				room.getName(),
+				gameService.isEnd((int)room.getId()))
+			)
+			.collect(Collectors.toList());
```

Comment:
```suggestion
		List<RoomResponseDto> rooms = roomService.findAll();
```
서비스에서 모두 처리하고, 컨트롤러는 데이터를 핸들링하는 역할만 하도록 해볼까요?

또 매번 룸의 갯수만큼 `isEnd()`를 호출하고 있어서 room의 갯수만큼 쿼리가 날라가겠군요?

Reply:
서비스에서 Dto를 반환하도록 변경하였습니다. 그리고 `Room `테이블에 `end `칼럼을 추가하여 `Room`을 조회했을 때 한 번에 종료됨을 알 수 있도록 변경했습니다

Reply:
모든 Room을 조회 후, room객체에서 isEnd() 메서드를 호출하긴 어려웠을까요?
필드 추가가 꼭 필요한 상황이었을까요?

Reply:
게임이 끝났는지에 대한 여부는 `Board`가 판단할 수 있습니다. 만약 `Room`이 `isEnd()`를 호출하려면 `Board`에 대한 의존성이 있어야 합니다. 그럼 `Room`을` findAll()` 했을 경우 `Room`들과 `Room`에 매핑된 `Board `및 `Piece`까지 다 긁어와야 합니다.

`Room`객체가 `Board`와 협력하여 필드 추가 없이 `isEnd()`를 사용하는 것이 더 객체지향스럽다고는 느끼고 있습니다. 그런데 `Room`을 찾아왔을 때 `Board`와 `Piece`까지 다 긁어오는 게 괜찮을까요?

제 얕은 지식으로는 나중에 JPA를 쓰면 `lazyloading`을 할 수 있다고는 하지만 그런 경우를 대비해서 `Room`이 `Board`를 가지도록 해서 `isEnd()`를 부르는 게 맞을까요?

Reply:
약간의 트레이드 오프가 있을 것 같아요.

> 그런데 Room을 찾아왔을 때 Board와 Piece까지 다 긁어오는 게 괜찮을까요?

지금정도의 규모에서는 큰 성능이슈는 없을 것 같아요.

물론 필드를 추가하는게 속도측면에서 좋을 수도 있지만, 몇가지 단점도 있다고 생각해요.

- 데이터와 비즈니스의 값이 다를 가능성이 생김
- 컬럼을 추가하는데의 비용